main/streams/cast.c

```diff
@@ -46,7 +46,14 @@ typedef struct {
 
 FILE *fopencookie(void *cookie, const char *mode, COOKIE_IO_FUNCTIONS_T *funcs)
 {
-	return funopen(cookie, funcs->reader, funcs->writer, funcs->seeker, funcs->closer);
+	FILE *file = funopen(cookie, funcs->reader, funcs->writer, funcs->seeker, funcs->closer);
+	if (file) {
+		/* Buffering of FILE handles is stateful.
+		 * A bailout during these can corrupt the state of the FILE handle
+		 * and cause memory corruption errors. See GH-11078. */
+		setvbuf(file, NULL, _IONBF, 0);
+	}
```

Comment:
Why is this not limited to MacOS / BSD if it's not an issue on Linux? This might also require some testing to be really sure that there's no perf impact.

Reply:
> Why is this not limited to MacOS / BSD if it's not an issue on Linux?

Consistency mainly

> This might also require some testing to be really sure that there's no perf impact.

I can only test on a mac that's awfully slow, or BSD under virtualization, so getting accurate performance numbers would be difficult for me.

Reply:
I'm mainly asking because of Linux as the buffering is now disabled there as well. Obviously fixing the issue is more important here so BSD based platforms (including MacOS) are out of question. I think if there's potentially perf impact on Linux, it should be measured carefully. I would probably prefer just to apply it for platforms where it's needed to reduce any such risk. At least I would start with that and potentially use it for Linux as well later but only for master if some testing show no impact.

Reply:
I'm happy to do MacOS testing on both Intel and ARM. Can you provide a benchmark script or suggest which test cases to run?

Reply:
@bukka see also https://github.com/php/php-src/issues/11078#issuecomment-2157128428

Reply:
Ah I completely missed that from looking just on this diff. I can see that now when looking to the whole cast.c. All good then.